internal/app/app.go

```diff
@@ -0,0 +1,55 @@
+package app
+
+import (
+	"PersonalAccountAPI/config"
+	"PersonalAccountAPI/database"
+	"PersonalAccountAPI/internal/cache"
+	"PersonalAccountAPI/internal/handler"
+	"PersonalAccountAPI/internal/metrics"
+	"PersonalAccountAPI/internal/repository"
+	"PersonalAccountAPI/internal/storage"
+	"PersonalAccountAPI/internal/uploading"
+	"PersonalAccountAPI/internal/usecase"
+	"PersonalAccountAPI/internal/workers"
+	"context"
+	"time"
+
+	"github.com/go-faster/errors"
+)
+
+func Run() error {
+	cfg, err := config.LoadConfig()
+	if err != nil {
+		return errors.Wrap(err, "Run config.LoadConfig: Failed to load config")
+	}
+
+	conn, err := storage.GetConnectDB(cfg.GetDSN())
+	if err != nil {
+		return errors.Wrap(err, "Run storage.GetConnectDB")
+	}
+	defer conn.Close(context.Background())
+
+	if err := database.Migrate(cfg.GetDSN()); err != nil {
+		return errors.Wrap(err, "Run database.Migrate")
+	}
+
+	s3Client, err := storage.InitS3Client(cfg.GetS3Config())
+	if err != nil {
+		return errors.Wrap(err, "Run storage.InitS3Client")
+	}
+
+	userRepository := repository.New(conn)
+	userProvider := usecase.New(userRepository)
+	cacheProvider := cache.New(userProvider, cfg.GetCacheTTL()*time.Second)
```

Comment:
Твоя логика hanlder -> cache -> usecase -> repo

Валидация, взаимодействие с другими сервисами, одним словом бизнес-логика идёт на слое usecase.
Если тебе пришёл невалидный запрос, то ты пойдёшь в кэш доставать данные по невалидным данным, это некорректно, после handler должен идти usecase, уже после usecase идёт cache, а затем база.

Корректно
handler -> usecase -> cache -> repo
```suggestion
    ...
	userRepository := repository.New(conn)
    cacheProvider := cache.New(userRepository, cfg.GetCacheTTL()*time.Second)
	userProvider := usecase.New(cacheProvider)
    ...
```
Контракты переделать тоже надо будет

Reply:
done